Bump Helm components

meyskens opened the pull request:
# Bump Helm components

Updates most Helm components in https://github.com/kinvolk/lokomotive/issues/1443


STILL WIP

Commit:
Update Kubernetes to v1.20.5

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update Etcd to v3.4.15

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update cert-manager to v1.2.0

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Correct the cert-manager update instructions

They seem to have gotten out of date and refering to an older file
structure. This makes them correct again. As well as fixes the spelling
of cert-manager.

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update dex to v2.28.1

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update OpenEBS to v2.7.0

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update the OpenEBS update instructions

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

surajssd commented on assets/charts/control-plane/kube-apiserver/Chart.yaml:
```diff
@@ -15,4 +15,4 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 version: 0.1.3
```
Also bump the chart version.

surajssd commented on assets/charts/control-plane/kubelet/Chart.yaml:
```diff
@@ -15,4 +15,4 @@ type: application
 # to the chart and its templates, including the app version.
 version: 0.1.3
-appVersion: v1.20.4
+appVersion: v1.20.5
```
Also bump the chart version.

surajssd commented on assets/charts/control-plane/kubernetes/Chart.yaml:
```diff
@@ -15,4 +15,4 @@ type: application
 # to the chart and its templates, including the app version.
 version: 0.1.3
-appVersion: v1.20.4
+appVersion: v1.20.5
```
Also bump the chart version.

surajssd commented on assets/charts/control-plane/kubelet/container-image/Dockerfile:
```diff
@@ -1,3 +1,3 @@
-FROM quay.io/poseidon/kubelet:v1.20.4
+FROM quay.io/poseidon/kubelet:v1.20.5
```
This image has to be built and pushed for the CI to pass.

meyskens replied:
Is pushed now :)

surajssd commented on assets/charts/components/dex/Chart.yaml:
```diff
@@ -20,4 +20,4 @@ version: 0.2.0
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
-appVersion: v2.27.0
+appVersion: v2.28.1
```
Also update the chart version.

surajssd commented on docs/development/updating-components.md:
```diff
@@ -81,12 +81,13 @@ helm fetch --untar --untardir ./ stable/metrics-server
 helm fetch --untar --untardir ./ openebs/openebs
-git checkout openebs/crds/storagepoolclaims.yaml
+mv openebs openebs-operator
+git checkout openebs-operator/crds/storagepoolclaims.yaml
```
With openebs update the update doc also needs update: https://github.com/kinvolk/lokomotive/blob/master/docs/how-to-guides/update-openebs.md.

meyskens replied:
done :heavy_check_mark:

Commit:
Update Rook to v1.5.9

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Add Rook update instructions

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update aws-ebs-csi-driver to v0.9.14

This also fixes Unit tests for aws ebs csi driver
The new chart adds new default tolerations the tests did not expect.

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Add update instructions for aws-ebs-csi-driver

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update Linkerd to v2.10.0

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Add Linkerd update instructions

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
left behind linkerd

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update Istio to  1.9.2

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Add Istio upgrade instructions

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update the update OpenEBS doc for the new versions

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update node-problem-detector to v0.8.7

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Document updating node-problem-detector

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update external-dns to v0.7.6

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Document how to update external-dns

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update Velero to v2.17.1

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Document how to update Velero

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Update cluster-autoscaler to 9.9.2

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

Commit:
Document how to update cluster-autoscaler

Signed-off-by: Maartje Eyskens <maartje@kinvolk.io>

ipochi commented on assets/terraform-modules/aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl:
```diff
@@ -149,7 +149,7 @@ storage:
       contents:
         inline: |
-          IMAGE_TAG=v3.4.14
+          IMAGE_TAG=v3.4.15
```
It would be nice if updating Kubernetes and etcd were moved to a  separate PR so that this Pr only focuses on updating the Lokomotive components.
